### PR TITLE
First pass in fixing createToken stripe not showing errors

### PR DESCRIPTION
**Pull request by ashkan18**
https://artsyproduct.atlassian.net/browse/PURCHASE-719

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 5b63f73cf4149a1667d642d6432d557c207d96b8

https://deploy-preview-1660--artsy-reaction.netlify.com

**Comment by ashkan18** [comment]
@ds300, me and @sweir27 started looking at this yesterday, we kind of got stuck on getting the test pass. The problem seems to be, current test mock the `createToken` method to return results but in this case we want to mock it as promise. Something like
```js
stripeMock.createToken.mockImplementationOnce(() => Promise.resolve(jest.fn()))
```
but still tests are failing with some weird errors. When had a chance can you take a look and see if there is something we can do to solve this. thanks

**Comment by ds300** [comment]
@ashkan18 haven't tested but it should probably be

```ts
stripeMock.createToken.mockImplementationOnce(() => Promise.reject(new Error("couldn't create token :(")))
```

**Comment by ds300** [comment]
@ashkan18 still working on this. So so close to finishing. This close 👌 but need to step out for a few hours. Will push what I have anyway.

**Comment by ashkan18** [comment]
THAAAAAAANKS for looking into this!

**Comment by ds300** [comment]
err accidentally made a new branch, will fix later and push it here https://github.com/ds300/reaction/tree/fix/payemnt-issue

**Comment by ds300** [comment]
@ashkan18 all done. The main problem was that promises were being mocked in too minimal a way in this test file, so I rewrote it to use actual promises that can be chained and support `catch` etc.

@damassi does the change to the sentry init logic make sense to you? It was being setup during tests because `sd.NODE_ENV` was `undefined`, which was overwriting jest's timers mocks somehow.

**Comment by yuki24** [comment]
Nice! Thanks for catching this!

**Comment by ashkan18** [comment]
this is awesome! thanks for dealing with this headache friendly issue :)

File: src/Apps/Order/Routes/__tests__/Payment.test.tsx [hunk 1]
```diff
@@ -282,25 +278,45 @@ describe("Payment", () => {
...
+    stripeMock.createToken.mockReturnValue(
+      Promise.reject(new Error("something failed"))
+    )
+    const paymentRoute = getWrapper(testProps)
+    fillAddressForm(paymentRoute, validAddress)
     paymentRoute.find(ContinueButton).simulate("click")
 
+    // flush promise queue
+    await Promise.resolve()
+    await Promise.resolve()
```
Comment (sweir27) [inline src/Apps/Order/Routes/__tests__/Payment.test.tsx]: Why do we have this twice?

Comment (ds300) [reply src/Apps/Order/Routes/__tests__/Payment.test.tsx]: jest has an internal promise queue. when you do 

    await Promise.resolve()

it pushes that new promise to the end of the queue and execution continues when the queue gets emptied up to and including that new promise.

However, if during that time more promises get added to the queue, then those will still be unresolved. If your test depends on those extra promises being resolved and some effects happening, you will need to flush the queue again. And maybe again. And so on. Depends how long your promise chains get. 

Something like this https://github.com/facebook/jest/issues/2157#issuecomment-279171856 might mitigate that. Worth trying!

**Pull request merged**